### PR TITLE
feat: add getMeetingInfo to TeamsInfo

### LINKED ISSUE #3723: Teams - get meeting details (JS)
See [parent](https://github.com/microsoft/botframework-sdk/issues/6304)

### PR DESCRIPTION
**Pull request by mdrichardson**
Fixes #3723 

## Description
Adds a `getMeetingInfo` method to `TeamsInfo`.

## Notes

Bots implementing this must add this to their `manifest.json`:

```json
"webApplicationInfo": {
        "id": "<MicrosoftAppId>",
        "resource": "<urlToApp>",
        "applicationPermissions": [
            "OnlineMeeting.ReadBasic.Chat"
        ]
    }
```

Per Teams, this is currently only available for:

1. Tenets who have the RSC permission for `OnlineMeeting.ReadBasic.Chat` set to Allowed. This won't be available for the Graph API permissions set until 6/17, or
2. Developers with the bot AppId whitelisted by the Teams Team (currently only available for Microsoft partner teams

## Testing
image

### CODE REVIEW

File: libraries/botframework-schema/src/teams/index.ts [hunk 1]
```diff
@@ -1687,3 +1687,57 @@ export type Type3 = MessagingExtensionResultType;
...
+     */
+    msGraphResourceId: string;
+    /**
+     * @member {Date} [scheduledStartTime] The meeting's scheduled start time, in UTC.
+     */
+    scheduledStartTime: Date;
+    /**
+     * @member {Date} [scheduledEndTime] The meeting's scheduled end time, in UTC.
+     */
+    scheduledEndTime: Date;
```
Comment (mdrichardson) [inline libraries/botframework-schema/src/teams/index.ts:1711]: Like [.NET](https://github.com/microsoft/botbuilder-dotnet/pull/5655#discussion_r649444421), I'm open to changing these to `string`.

File: libraries/botframework-connector/src/teams/operations/teams.ts [hunk 2]
```diff
@@ -181,6 +182,51 @@ export class Teams {
...
+        meetingId: string,
+        options?: msRest.RequestOptionsBase | msRest.ServiceCallback<TeamDetails>
+    ): Promise<Models.TeamsMeetingInfoResponse>;
+    /**
+     * @param meetingId Meeting Id, encoded as a BASE64 string.
+     * @param callback The callback
+     */
+     fetchMeetingDetails(
+        meetingId: string,
+        callback: msRest.ServiceCallback<TeamsMeetingInfo>
```
Comment (mdrichardson) [inline libraries/botframework-connector/src/teams/operations/teams.ts:204]: `...Details` returns type `...Info`. [Discuss here](https://github.com/microsoft/botbuilder-dotnet/pull/5655#discussion_r649540713)

File: libraries/botbuilder/src/teamsInfo.ts [hunk 3]
```diff
@@ -81,6 +82,31 @@ export class TeamsInfo {
...
+    public static async getMeetingInfo(context: TurnContext, meetingId?: string): Promise<TeamsMeetingInfo> {
+        if (!context) {
+            throw new Error('context is required.');
+        }
+
+        const activity = context.activity;
+
+        if (meetingId == null) {
+            const meeting = teamsGetTeamMeetingInfo(activity);
+            meetingId = meeting ? meeting.id : undefined;
```
Comment (joshgummersall) [inline libraries/botbuilder/src/teamsInfo.ts:100]: You can write this as `meetingId = meeting?.id;`

File: libraries/botbuilder/tests/teamsInfo.test.js [hunk 4]
```diff
@@ -661,6 +661,96 @@ describe('TeamsInfo', function () {
...
+
+            const fetchedDetails = await TeamsInfo.getMeetingInfo(context, details.details.id);
+
+            assert(fetchOauthToken.isDone());
+            assert(fetchExpectation.isDone());
+
+            assert.deepStrictEqual(fetchedDetails, details);
+        });
+
+        it('should throw error for missing context', async function () {
```
Comment (joshgummersall) [inline libraries/botbuilder/tests/teamsInfo.test.js:749]: Worth adding a test for an undefined or null `meetingId`?

File: libraries/botframework-schema/src/teams/index.ts (hunk 1 above)
Comment (joshgummersall) [reply libraries/botframework-schema/src/teams/index.ts:1711]: Does `msrest` parse JSON strings into a `Date` object?

**Review by joshgummersall — COMMENTED** [review]
Couple quick questions, otherwise this looks good.

File: libraries/botframework-schema/src/teams/index.ts (hunk 1 above)
Comment (mdrichardson) [reply libraries/botframework-schema/src/teams/index.ts:1711]: image

Sure does. (works for .NET, too)

File: libraries/botbuilder/src/teamsInfo.ts (hunk 3 above)
Comment (mdrichardson) [reply libraries/botbuilder/src/teamsInfo.ts:100]: Ah yeah, I should've caught that. Fixed it here as well as the place I copy/pasted that piece from.

File: libraries/botbuilder/tests/teamsInfo.test.js (hunk 4 above)
Comment (mdrichardson) [reply libraries/botbuilder/tests/teamsInfo.test.js:749]: Sort of. It can pull from the turnContext, but I'll add a test to ensure it throws if it can't find it in the turnContext.

**Review by joshgummersall — APPROVED** [review]
